volume/parser.go

```diff
@@ -0,0 +1,43 @@
+package volume
+
+import (
+	"runtime"
+
+	"github.com/docker/docker/api/types/mount"
+)
+
+const (
+	// PlatformLinux is the same as runtime.GOOS on linux
+	PlatformLinux = "linux"
+	// PlatformWindows is the same as runtime.GOOS on windows
+	PlatformWindows = "windows"
+)
+
+// Parser represents a platform specific parser for mount expressions
+type Parser interface {
```

Comment:
Yes I agree. I started with 2 methods, and I progressively had to add a lot of stuff in it...